engine/src/test/java/io/zeebe/engine/processor/workflow/subprocess/EventSubprocessTest.java

```diff
@@ -0,0 +1,217 @@
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.engine.processor.workflow.subprocess;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.tuple;
+
+import io.zeebe.engine.util.EngineRule;
+import io.zeebe.model.bpmn.Bpmn;
+import io.zeebe.model.bpmn.BpmnModelInstance;
+import io.zeebe.model.bpmn.builder.EmbeddedSubProcessBuilder;
+import io.zeebe.model.bpmn.builder.ProcessBuilder;
+import io.zeebe.protocol.record.Assertions;
+import io.zeebe.protocol.record.Record;
+import io.zeebe.protocol.record.intent.WorkflowInstanceIntent;
+import io.zeebe.protocol.record.value.BpmnElementType;
+import io.zeebe.protocol.record.value.WorkflowInstanceRecordValue;
+import io.zeebe.test.util.record.RecordingExporter;
+import io.zeebe.test.util.record.RecordingExporterTestWatcher;
+import java.util.List;
+import java.util.Map;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class EventSubprocessTest {
+  @ClassRule public static final EngineRule ENGINE = EngineRule.singlePartition();
+
+  @Rule
+  public final RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @Test
+  public void shouldTriggerEventSubprocess() {
+    // given
+    final BpmnModelInstance model = createEventSubProc();
+
+    // when
+    final long workflowKey =
+        ENGINE
+            .deployment()
+            .withXmlResource(model)
+            .deploy()
+            .getValue()
+            .getDeployedWorkflows()
+            .get(0)
+            .getWorkflowKey();
+    final long workflowInstanceKey =
+        ENGINE.workflowInstance().ofBpmnProcessId("proc").withVariable("innerKey", "123").create();
+
+    // then
+    final Record<WorkflowInstanceRecordValue> timerTriggered =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.EVENT_OCCURRED).getFirst();
+    Assertions.assertThat(timerTriggered.getValue())
+        .hasWorkflowKey(workflowKey)
+        .hasWorkflowInstanceKey(workflowInstanceKey)
+        .hasBpmnElementType(BpmnElementType.START_EVENT)
+        .hasElementId("event_sub_start");
+
+    assertEventSubprocessLifecycle(timerTriggered);
+  }
+
+  @Test
+  public void shouldTriggerNestedEventSubproc() {
+    // given
+    final BpmnModelInstance model = createNestedEventSubproc();
+    ENGINE.deployment().withXmlResource(model).deploy();
+
+    // when
+    ENGINE.workflowInstance().ofBpmnProcessId("proc").create();
+
+    // then
+    final Record<WorkflowInstanceRecordValue> timerTriggered =
+        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.EVENT_OCCURRED).getFirst();
+
+    assertEventSubprocessLifecycle(timerTriggered);
+  }
+
+  @Test
+  public void shouldPropagateVariablesToEventSubprocess() {
```

Comment:
It seems that the test doesn't verify that the variables can be accessed inside the event subprocess.